src/service/cmake/CodeCoverage.cmake

```diff
@@ -239,7 +239,7 @@ function(setup_target_for_coverage_lcov)
     if(DEFINED Coverage_BASE_DIRECTORY)
         get_filename_component(BASEDIR ${Coverage_BASE_DIRECTORY} ABSOLUTE)
     else()
-        set(BASEDIR ${PROJECT_SOURCE_DIR})
+        set(BASEDIR ../../..)
```

Comment:
Just for the record, not a fan of relative paths (I use them as well, but I don't like them)

Reply:
yes, I wanted to be able to compile the subprojects independently as well as from the top-level cmake, hence `$PROJECT_SOURCE_DIR` can point to different directories.
I can replace it by `<PROJECT-NAME>_SOURCE_DIR` to get rid of the relative path, but hardcoding the subproject name also feels wrong. I'll leave it for now. Coverage is not used at the moment anyway. Setting up tests and coverage reporting will be a separate PR anyway.